src/main/java/wolox/training/models/Book.java

```diff
@@ -57,6 +59,8 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        Preconditions.checkArgument(genre != null && !genre.isEmpty()
```

Comment:
Check out [this method by Guava](https://guava.dev/releases/27.1-jre/api/docs/com/google/common/base/Strings.html#isNullOrEmpty-java.lang.String- )
It's cool for null and empty checks with Strings 😄 (just a suggestion for the future)